core/jvmTest/src/kotlinx/serialization/SerializeFlatTest.kt

```diff
@@ -297,4 +298,34 @@ class SerializeFlatTest() {
             if (step != 7) fail("@$step: INP FAIL")
         }
     }
+
+    class StringEncoder(
+            private val expected: String,
+            override val serializersModule: SerializersModule = EmptySerializersModule
+    ): AbstractEncoder() {
+        override fun encodeString(value: String) {
+            assertEquals(expected, value)
+        }
+    }
+
+    class StringDecoder<T>(
+            private val expected: T,
+            override val serializersModule: SerializersModule = EmptySerializersModule
+    ): AbstractDecoder() {
+        override fun decodeElementIndex(descriptor: SerialDescriptor): Int {
+            assertEquals(1, descriptor.elementsCount)
+            return 1
+        }
+        override fun decodeString(): String = expected.toString()
+    }
+
+    @Test
+    fun testUUID() {
```

Comment:
This method uses a reflective approach to resolve the serializer. Also add one that uses `UUID.serializer`, You will find that will not work currently.